shell/browser/osr/osr_render_widget_host_view.cc

```diff
@@ -233,17 +233,32 @@ void OffScreenRenderWidgetHostView::OnLocalSurfaceIdChanged(
 }
 
 OffScreenRenderWidgetHostView::~OffScreenRenderWidgetHostView() {
-  render_widget_host_->render_frame_metadata_provider()->RemoveObserver(this);
+  ReleaseCompositor();
+  root_layer_.reset(nullptr);
+
+  DCHECK(!parent_host_view_);
+  DCHECK(!popup_host_view_);
+  DCHECK(!child_host_view_);
+  DCHECK(guest_host_views_.empty());
+}
+
+void OffScreenRenderWidgetHostView::ReleaseCompositor() {
+  if (!compositor_) {
+    return;  // already released
+  }
 
   // Marking the DelegatedFrameHost as removed from the window hierarchy is
   // necessary to remove all connections to its old ui::Compositor.
-  if (is_showing_)
+  if (is_showing_) {
     delegated_frame_host_->WasHidden(
         content::DelegatedFrameHost::HiddenCause::kOther);
+  }
+
   delegated_frame_host_->DetachFromCompositor();
+  delegated_frame_host_.reset(nullptr);
```

Comment:
Does clearing `delegated_frame_host_` here cause problems in the destructor where we dereference it, e.g. [here](https://github.com/electron/electron/blob/main/shell/browser/osr/osr_render_widget_host_view.cc#L243)?

Reply:
@ckerr maybe it's showing up weirdly in the diff view, but this PR moved that logic into `OffScreenRenderWidgetHostView::ReleaseCompositor()`

Reply:
Nah it's showing up fine, I just didn't see it. Thanks! :+1: